packages/contracts/src/core/permission/PermissionManager.sol

```diff
@@ -109,6 +219,210 @@ abstract contract PermissionManager is Initializable {
+    /// @param _where The target contract to revoke or give permissions on.
+    /// @param _permissionId The permission to check the permissions for.
+    /// @param _operation The operation used to check ownership. The use of the PermissionLib.Operation type here instead of PM.Option simplifies the code (see applyTarget methods).
+    modifier ownerAuth(
+        address _where,
+        bytes32 _permissionId,
+        PermissionLib.Operation _operation
+    ) {
+        Permission storage permission = permissions[permissionHash(_where, _permissionId)];
+
+        if (_isPermissionFrozen(permission)) {
+            revert PermissionFrozen(_where, _permissionId);
+        }
+
+        if (!_checkOwner(permission, msg.sender, _operation, _isRoot(msg.sender))) {
+            revert Unauthorized(_where, msg.sender, _permissionId);
+        }
+
+        _;
+    }
+
+    /// @dev Requires the `ROOT_PERMISSION_ID` permission.
+    /// @param _where The address of the target contract for which `_who` receives permission.
+    /// @param _permissionIdOrSelector The permission hash or function selector used for this permission.
+    /// @param _owner The initial owner of this newly created permission.
+    /// @param _whos The addresses of the target contracts for which `_who` receives permission.
+    function createPermission(
+        address _where,
+        bytes32 _permissionIdOrSelector,
+        address _owner,
+        address[] calldata _whos
+    ) external auth(ROOT_PERMISSION_ID) {
+        _createPermission(_where, _permissionIdOrSelector, _owner, _whos);
+    }
+
+    /// @notice Function to delegate specific flags of a permission.
+    /// @param _where The address of the target contract for which `_who` receives permission.
+    /// @param _permissionIdOrSelector The permission hash or function selector used for this permission.
+    /// @param _delegatee The addresses who gets the permission delegated.
+    /// @param _flags The flags as uint256 the permission owner wants to give this specific delegatee.
+    function delegatePermission(
+        address _where,
+        bytes32 _permissionIdOrSelector,
+        address _delegatee,
+        uint256 _flags
+    ) public {
+        if (_flags == 0) {
+            revert FlagCanNotBeZero();
+        }
+
+        bytes32 permHash = permissionHash(_where, _permissionIdOrSelector);
+        Permission storage permission = permissions[permHash];
+
+        if (_isPermissionFrozen(permission)) {
+            revert PermissionFrozen(_where, _permissionIdOrSelector);
+        }
+
+        if (!_checkFlags(permission.owners[msg.sender], _flags)) {
+            revert UnauthorizedOwner(msg.sender, permission.owners[msg.sender], _flags);
+        }
+
+        uint256 newFlags = permission.delegations[_delegatee] | _flags;
+        permission.delegations[_delegatee] = newFlags;
+
+        emit PermissionDelegated(_where, _permissionIdOrSelector, _delegatee, newFlags);
```

Comment:
if `newFlags` is zero (all new flags being added were already added before) why store and emit the event again instead of just revert or just do nothing to avoid paying gas?